convert2rhel/unit_tests/cert_test.py

```diff
@@ -98,30 +98,38 @@ def test_install_cert(self):
         shutil.rmtree(unit_tests.TMP_DIR)
 
 
+@pytest.mark.cert_filename("filename")
+def test_remove_cert(caplog, system_cert_with_target_path):
+    cert_file_path = system_cert_with_target_path._target_cert_path
+    with open(cert_file_path, "wb") as cert_file:
+        cert_file.write(b"some content")
+
+    system_cert_with_target_path.remove()
+
+    assert "/filename removed" in caplog.messages[-1]
+
+
 @pytest.mark.parametrize(
     (
-        "create_file",
+        "error_condition",
         "expected_text_in_logs",
     ),
     (
-        (True, "/filename removed"),
-        (False, "No such file or directory"),
+        (
+            OSError(2, "No such file or directory"),
+            "No such file or directory",
+        ),
+        (OSError(13, "[Errno 13] Permission denied: '/tmpdir/certfile'"), "Permission denied:"),
     ),
 )
-def test_remove_cert(
-    monkeypatch,
-    create_file,
-    expected_text_in_logs,
-    caplog,
-    tmpdir,
+def test_remove_cert_error_conditions(
+    error_condition, expected_text_in_logs, caplog, monkeypatch, system_cert_with_target_path
 ):
-    tmp_file = tmpdir / "filename"
-    if create_file:
-        tmp_file.write(b"some content")
+    def fake_os_remove(path):
+        raise error_condition
+
+    monkeypatch.setattr(os, "remove", fake_os_remove)
 
-    monkeypatch.setattr(cert.SystemCert, "_get_cert", value=mock.Mock(return_value=("anything", "anything")))
-    monkeypatch.setattr(cert.SystemCert, "_get_target_cert_path", value=mock.Mock(return_value=str(tmp_file)))
+    system_cert_with_target_path.remove()
 
-    sys_cert = cert.SystemCert()
-    sys_cert.remove()
-    assert expected_text_in_logs in caplog.messages[-1]
+    assert expected_text_in_logs != caplog.messages[-1]
```

Comment:
I was wondering about these tests, I think that we could tweak it a little bit and instead of checking against a specific message, we could check for the message we want to achieve. 

Let me give a more concrete example here: 

Right now, we are asserting that a certain message does not appear after removing the certificates, although this is usually an okay case, I believe for this test specific we should do the inverse and check that we have the message we want. This way I feel that we can be more assertive and be sure that the test/code is working properly. 

## What are we doing right now

```python
@pytest.mark.parametrize(
    (
        "create_file",
        "error_condition",
        "expected_text_in_logs",
    ),
    (
        (True, "/filename removed"),
        (False, "No such file or directory"),
        (
            OSError(2, "No such file or directory"),
            "No such file or directory",
        ),
        (OSError(13, "[Errno 13] Permission denied: '/tmpdir/certfile'"), "Permission denied:"),
    ),
)
...
assert expected_text_in_logs != caplog.messages[-1]
```
The parameter `expected_text_in_logs` doesn't have a really good usage here, since it could be anything like `"aaa"` and the tests would pass without a problem (And, as I said, this can be okay in some scenarios). But we are not necessarily asserting that the message being sent to the user is the correct one, right?  We are just assuming that whatever it outputs is different from the `expected_text_in_logs` message. 

## What we could do (suggestion) 

I believe to change this, we could only do a minimal change to the test, and instead of checking that the `expected_text_in_logs` is not present in the log, we can check that the `expected_text_in_logs` is actually present. This way, we can be sure that, for example, if the cert was not located in the system during the removal, we are outputting the right message to the user with a proper message. 

So, my suggestion is something like this: 

```python
@pytest.mark.parametrize(
    (
        "create_file",
        "error_condition",
        "expected_text_in_logs",
    ),
    (
        (True, "/filename removed"),
        (False, "No such file or directory"),
        (
            OSError(2, "No such file or directory"),
            "No RHSM certificates found to be removed",
        ),
        (OSError(13, "[Errno 13] Permission denied: '/tmpdir/certfile'"), "OSError(13): Permission denied: '/tmpdir/certfile'"),
    ),
)
...
assert expected_text_in_logs in caplog.messages[-1]
```

Also, the test becomes more complete since the issue in Jira says that the message `ERROR - OSError(2): No such file or directory` should not appear in that cert removal. 

----

What do you think about that? It's just a suggestion that I thought it was valid to point out!

Reply:
My applogies for taking so long to reply on this comment! Thank you so much for sharing this very insightful comments on this PR! It is very helpful in making sure that I understand what it is that you are saying for this comment!